### ISSUE TITLE
Review `webpack`'s new governance model -- Impact project.

### ISSUE THREAD

**Issue by ovflowd** [openjs-foundation/cross-project-council#1363]
Hey folks 👋 as discussed in a few prior CPC meetings, `webpack` was at risk of being downplayed from `Impact` to `At Large` due to a missing governance model and insufficient maintainers.

After the meeting, I synced with @rginn and joined `webpack` as one of the maintainers. We immediately started working on rewriting `webpack`'s governance mode, to be more in line with current standards of other `Impact` projects, and we installed a TSC (Technical Steering Committee) for `webpack`; Which will have at the moment more than five members (We are assessing who wants to join the TSC and whatnot)

We reached an internal consensus that the governance model is almost ready for review by the Foundation's CPC and the Board since it has undergone a massive change.

The governance model can be viewed at https://github.com/webpack/webpack-governance; at the moment, initial feedback from the CPC is just being gathered, which can be found here.

Once we believe we are ready for a final review, a Pull Request will be made against https://github.com/webpack/webpack, updating the GOVERNANCE.md file and README.md file to reflect the new Governance Model and webpack TSC.

Hence, the ask is for the CPC to start analyzing, viewing, and reviewing (by adding comments here) the initial draft and raise awareness of the incoming governance model change, pending approval by the CPC and the Board.

**Comment by bensternthal** [comment]
@ovflowd My understanding is that the board does not review or approve a project's technical charter and that this is solely under the CPC. If you found something that states otherwise please let me know.

**Comment by ovflowd** [comment]
> @ovflowd My understanding is that the board does not review or approve a project's technical charter and that this is solely under the CPC. If you found something that states otherwise please let me know.

Oh good to know! Nope, I just assumed 🙇

**Comment by tobie** [comment]
As expressed in Slack, I'm a little bit confused as to what the request is here. Is this a request for input on the governance project at large? Is it a request to formally approve webpack's new charter?

If it's the former, I've expressed my concerns in slack about the complexity of the governance compared to the size of the project and would strongly advise simplifying it greatly and avoiding duplication of information across documents and repositories. But this is absolutely not mandatory; projects have a lot of leeway when it comes to establishing their internal governance, they just have to abide by the general principles of the foundation and have a charter that is formally approved an reviewed by the CPC.

If it's the latter, the CPC will want to review webpack's charter, not governance.md or a README. (I'd recommend basing the charter on the [template charter](https://github.com/openjs-foundation/cross-project-council/blob/main/PROJECT_CHARTER_TEMPLATE.md) available in the CPC's repo as it's going to lighten the review burden on the CPC significantly.)

@ovflowd would you mind clarifying what the intent of this issue and of the related PR (https://github.com/webpack/webpack/pull/18804) you linked in slack? I think this would be super helpful. Thanks!

**Comment by voxpelli** [comment]
> as discussed in a few prior CPC meetings, `webpack` was at risk of being downplayed from `Impact` to `At Large` due to a missing governance model and insufficient maintainers

My interpretation is that @ovflowd is requesting a review of whether this new governance model would remove/reduce the risk of `webpack` being downplayed from `Impact` to `At Large` – because else the work that @ovflowd and others has done has not been successful (and then hopefully there are some feedback on how it could be tweaked for it to become successful in achieving its goals)

@ovflowd Am I correct?

**Comment by ovflowd** [comment]
> @ovflowd Am I correct?

Yes! This is what originally motivated us for doing such work. In the end we realised the webpack project (somehow (?)) does not have a concrete/real charter at the moment (I wonder how they became part of OpenJS without that, or maybe there was at some point, or it is legacy enough)

But at least the current maintainers concur that there is no real governing model for the whole webpack org. Hence, we drafted a real charter to create webpack's TSC and a governing model document to reflect on current practices and future ones.

**Comment by tobie** [comment]
> We are formally requesting for the CPC to vote on approval of the new TSC charter for webpack. We are also requesting optional feedback on the new governance model and its wording.

Thanks for clarifying. Can you please link to the charter and be explicit that this is a formal charter approval request.

**Comment by ovflowd** [comment]
> > We are formally requesting for the CPC to vote on approval of the new TSC charter for webpack. We are also requesting optional feedback on the new governance model and its wording.
> 
> Thanks for clarifying. Can you please link to the charter and be explicit that this is a formal charter approval request.

Definitely! The actual chart file was created here: https://github.com/webpack/webpack-governance/blob/main/TSC_CHARTER.md

In this PR, we can address any change to the CHARTER and the rest of the governance repository: https://github.com/webpack/webpack-governance/pull/6

But ultimately, this https://github.com/webpack/webpack/pull/18804 is the PR that requires 👍 from the CPC for us to proceed with the adoption of the charter.

Right now, that webpack/webpack-governance repository holds no power as it is not referred anywhere.

**Comment by tobie** [comment]
> But ultimately, this https://github.com/webpack/webpack/pull/18804 is the PR that requires 👍 from the CPC for us to proceed with the adoption of the charter.

That's incorrect. As explained previously, the CPC will approve the charter, not this PR.

**Comment by ovflowd** [comment]
> > But ultimately, this [webpack/webpack#18804](https://github.com/webpack/webpack/pull/18804) is the PR that requires 👍 from the CPC for us to proceed with the adoption of the charter.
> 
> That's incorrect. As explained previously, the CPC will approve the charter, not this PR.

I thought that charter changes the CPC goes to the PR to approve said charter changes. I saw that happening elsewhere, like with the JSON schema org, is that not the case anymore?

**Comment by tobie** [comment]
https://github.com/webpack/webpack/pull/18804 updates the README and governance.md. The CPC approves a charter (see [Art. 5(7)](https://github.com/webpack/webpack/pull/18804)).

**Comment by ovflowd** [comment]
> https://github.com/webpack/webpack/pull/18804 updates the README and governance.md. The CPC approves a charter (see [Art. 5(7)](https://github.com/openjs-foundation/cross-project-council/blob/main/CPC-CHARTER.md#section-5-responsibilities-and-expectations-of-the-cpc)).

Gotcha, would the other PR I mentioned above that is directly towards the webpack/webpack-governance repository then count as the proper place?

Or would no interactions in any PR be needed? Just here between ourselves? I know Im not a voting member so Im not part of the people that vote.

**Comment by tobie** [comment]
Ideally, the way this works is you have a PR containing only the charter chnages and that's what gets reviewed. Not sure how to manage this here now that the charter is already in a repo.

**Comment by ovflowd** [comment]
> Ideally, the way this works is you have a PR containing only the charter changes and that's what gets reviewed. Not sure how to manage this here now that the charter is already in a repo.

Noted, how else would it be reviewed if not in a repo? 🤔 like the changes be in a fork and then only added to this new repo after reviewed?

@evenstensberg do you believe we can rollback the changes to the main webpack/webpack-governance repo based from your fork then?

**Comment by tobie** [comment]
> how else would it be reviewed if not in a repo?

As a PR!

**Comment by tobie** [comment]
Concretely, what you could do is just create a PR that moves it so CHARTER.md.

Technically, our charters are project charters and not TSC charters, so CHARTER.md is a good place for it, actually.

**Comment by ovflowd** [comment]
> > how else would it be reviewed if not in a repo?
> 
> 
> 
> As a PR!

Sorry, my wording here was poor, I meant any PR needs to be against a repository; So I was asking if not as a PR. Then is it correct that the PR should be from the fork -> org repo?

**Comment by tobie** [comment]
The PR needs to be against the final destination of the charter, wherever that'll be.

**Comment by ovflowd** [comment]
I see, appreciate your guidance here. Ill sync with the team :)

**Comment by tobie** [comment]
Sounds great. I don't like making you do more work, here, but I do want to point out that you're going to get a lot of change requests the way the current charter is framed. I think you'd avoid yourself a lot of frustration if you reformatted it to meet the structure of the template charter I listed earlier. While we're not asking existing projects to update their charters, expectation is that new ones (or in this case, projects which don't have a charter yet) use the provided template as it makes everyone's life a lot easier.

**Comment by ovflowd** [comment]
> Sounds great. I don't like making you do more work here, but I do want to point out that you're going to get a lot of change requests the way the current charter is framed. I think you'd avoid yourself a lot of frustration if you reformatted it to meet the structure of the template charter I listed earlier. While we're not asking existing projects to update their charters, expectation is that new ones (or in this case, projects which don't have a charter yet) use the provided template as it makes everyone's life a lot easier (see reference to the charter template in the [onboarding checklist](https://github.com/openjs-foundation/cross-project-council/blob/main/PROJECT_PROGRESSION.md#onboarding-checklist)).

I genuinely thought we followed the template -- but Ill check with the team, as this requires even more work here...

**Comment by ovflowd** [comment]
Alrighty, the updates on the charter file actually look more trivial than I thought; That's definitely doable work 🙌

**Comment by tobie** [comment]
**Notes from today's CPC call:**

  * Tobie gave input on how to structure charter/governance based off of templates; Claudio and team will come back with updates

**Comment by ovflowd** [comment]
CTAs:

- Sync with @evenstensberg on Wednesday for updating the charter proposal
- Reset the governance repository that should not have the charter merged yet
- Open a PR towards the governance repository only containing the charter
- Request the CPC to review the charter
- Other changes and documents to do in follow-up PRs
  - For example, FUNDING documentation, verify first with WebdriverIO and Mocha as they also have more up to date templates.

**Comment by tobie** [comment]
**Notes from today's CPC call:**

See @ovflowd's notes above.

**Comment by ovflowd** [comment]
Hey @openjs-foundation/cpc the CHARTER is ready to be reviewed: https://github.com/webpack/governance/pull/1

**Comment by tobie** [comment]
**Notes from this week's CPC call:**
 
* Claudio commented a couple weeks ago that governance is ready again
